hybrid/power_source.py

```diff
@@ -238,27 +241,35 @@ def simulate_financials(self, interconnect_kw: float, project_life: int):
         if self.system_capacity_kw <= 0:
             return
 
-        self._financial_model.FinancialParameters.analysis_period = project_life
-        self._financial_model.Lifetime.system_use_lifetime_output = 1 if project_life > 1 else 0
-        self._financial_model.Revenue.ppa_soln_mode = 1
+        self._financial_model.value('analysis_period', project_life)
+        self._financial_model.value('system_use_lifetime_output', 1 if project_life > 1 else 0)
+        self._financial_model.value('ppa_soln_mode', 1)
 
         # try to copy over system_model's generation_profile to the financial_model
-        if len(self._financial_model.SystemOutput.gen) == 1:
-            if len(self.generation_profile) == self.site.n_timesteps:
-                self._financial_model.SystemOutput.gen = self.generation_profile
+        if len(self._financial_model.value('gen')) == 1:
+            if len(self.generation_profile) == self.site.n_timesteps or \
+              len(self.generation_profile) == self.site.n_timesteps * project_life:
+                self._financial_model.value('gen', self.generation_profile)
             else:
                 raise RuntimeError(f"simulate_financials error: generation profile of len {self.site.n_timesteps} required")
 
-        if len(self._financial_model.SystemOutput.gen) == self.site.n_timesteps:
-            single_year_gen = self._financial_model.SystemOutput.gen
-            self._financial_model.SystemOutput.gen = list(single_year_gen) * project_life
-
-        if self.name != "Grid":
-            self._financial_model.SystemOutput.system_pre_curtailment_kwac = self._system_model.value("gen") * project_life
-            self._financial_model.SystemOutput.annual_energy_pre_curtailment_ac = self._system_model.value("annual_energy")
-            # TODO: Should we use the nominal capacity function here?
+        if len(self._financial_model.value('gen')) == self.site.n_timesteps:
+            self._financial_model.value('gen', self._financial_model.value('gen') * project_life)
+        self._financial_model.value('system_pre_curtailment_kwac', self._financial_model.value('gen'))
+        self._financial_model.value('annual_energy_pre_curtailment_ac', self._system_model.value("annual_energy"))
+        # TODO: Should we use the nominal capacity function here?
         self.gen_max_feasible = self.calc_gen_max_feasible_kwh(interconnect_kw)
         self.capacity_credit_percent = self.calc_capacity_credit_percent(interconnect_kw)
+        if not isinstance(self._financial_model, Singleowner.Singleowner):
+            try:
+                self._financial_model.set_financial_inputs(flatten_dict(self._system_model.export()))
+            except:
+                raise NotImplementedError("Financial model cannot set its inputs.")
+        elif self.name!= "Grid" and isinstance(self._system_model, Pvsamv1.Pvsamv1):
```

Comment:
I feel like if it's possible, this should be moved to the detailed_pv_plant's `simulate_financials`:

```
super().simulate_financials()
pvsamv1 battery stuff
```

Reply:
You're right, that's much cleaner.